go/mysql/conn.go

```diff
@@ -464,35 +503,24 @@ func (c *Conn) startEphemeralPacket(length int) []byte {
 }
 
 // writeEphemeralPacket writes the packet that was allocated by
-// startEphemeralPacket. If 'direct' is set, we write to the
-// underlying connection directly, by-passing the write buffer.
-func (c *Conn) writeEphemeralPacket(direct bool) error {
+// startEphemeralPacket.
+func (c *Conn) writeEphemeralPacket() error {
 	defer c.recycleWritePacket()
 
-	var w io.Writer = c.writer
-	if direct {
-		w = c.conn
-	}
-
 	switch c.currentEphemeralPolicy {
 	case ephemeralWriteSingleBuffer:
 		// Write the allocated buffer as a single buffer.
 		// It has both header and data.
-		if n, err := w.Write(*c.currentEphemeralWriteBuffer); err != nil {
+		if n, err := c.getWriter().Write(*c.currentEphemeralWriteBuffer); err != nil {
 			return fmt.Errorf("Conn %v: Write(*c.currentEphemeralWriteBuffer) failed: %v", c.ID(), err)
```

Comment:
do you remember why there was no flush() path here?

Reply:
If `direct ` is true, then we directly write to c.conn. So, there's no need to flush. There is a subtle futuristic bug: if someone had previously invoked writeEphemeral with `false`, then something would be sitting in the write buffer, and the next call with `true` would skip those buffered items, and directly write the current packet to the conn.

The new code protects from this possibility.

Reply:
ah okay. thanks for explanation